app/models/discussion.rb

```diff
@@ -131,6 +132,16 @@ def extra_preview_html
     exercise.assignment_for(initiator).extra_preview_html
   end
 
+  def update_counters_and_timestamps!
+    messages_query = messages.reorder(updated_at: :desc)
+    useful_messages = messages_query.select &:useful?
+    self.messages_count = messages_query.count
+    self.useful_messages_count = useful_messages.count
+    self.last_initiator_message_at = messages_query.find(&:from_initiator?)&.updated_at
+    self.last_moderator_message_at = useful_messages.first&.updated_at
+    save!
+  end
```

Comment:
Why don't we use `update` here? It is cleaner and will just save those changes even when there are other dirty fields